src/test/java/seedu/address/model/good/GoodNameTest.java

```diff
@@ -17,11 +17,17 @@ public void constructor_null_throwsNullPointerException() {
         assertThrows(NullPointerException.class, () -> new GoodName(null));
     }
 
+
+    /*
+    @jd to fangsh: the constructor of GoodName doesn't perform any input validation.
+    Did you manage to get this test to throw an error?
+    Commenting this out until resolution
     @Test
     public void constructor_invalidName_throwsIllegalArgumentException() {
         String invalidName = "";
         assertThrows(IllegalArgumentException.class, () -> new GoodName(invalidName));
     }
+     */
```

Comment:
The validation in GoodName got deleted somewhere in your commits

Reply:
Yep, i see it at the top. Added it back in and now test passes!